core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -772,7 +772,7 @@ private[deploy] object Worker extends Logging {
     // more detail see SPARK-20989.
     val externalShuffleServiceEnabled = conf.get(config.SHUFFLE_SERVICE_ENABLED)
     val sparkWorkerInstances = scala.sys.env.getOrElse("SPARK_WORKER_INSTANCES", "1").toInt
-    require(externalShuffleServiceEnabled == false || sparkWorkerInstances <= 1,
+    require(!externalShuffleServiceEnabled || sparkWorkerInstances <= 1,
```

Comment:
I agree with removing `== true`. There is some minor argument against `!` vs `== false` as the `!` can be easier to overlook. Although it comes first, I suppose you could say. I'm neutral on those.

Reply:
It sounds reasonable, thanks